docs/watched literals.md

```diff
@@ -0,0 +1,21 @@
+## 2-Watched literals
+
+As you already know, `propagation` takes most time of the solving process.
+In order to reduce the time, we use 2-watched literals heuristic.
+> **Note:** it's one of the most useful and important heuristics for a good solver.
+
+The idea is to keep 2 pointers for each clause. These pointers "watch"
+on unassigned literals for clause. This helps us find unit clauses.
+If assigned literal is not watching clause then this clause can't become
+unit as it has 2 unassigned literals left. Otherwise, we go through all literals
+in clause and replace chosen literal with unassigned. If there are no new
+unassigned literals - clause becomes unit. So we don't waste time on
+watching all clauses and checking if the become unit. So we strongly 
+reduce the time of the longest solver's function. That's why this
+gives good improvement for the solving process.
+
+We've taken implementation used in MiniSat as it's pretty effective and hasn't
+changed a lot since that time.
```

Comment:
The fact that MiniSat stopped its evolution does not make any favor (i.e. cannot be used as an argument) for choosing its implementation as a reference one. Note that I'm not saying that MiniSat stagnation is bad — maybe it even good that MiniSat is frozen in time, so we can research it in its original form. However, you probably wanted to use as an argument the fact that any other 2-watched literal schemas are not very different from MiniSat's one, and are more or less of the same efficacy — then I agree, this is more-or-less correct. But in order to say it, you have to actually say it (!), including the mentioning of "others" and saying that "others" (approaches and/or authors) perform the similar manipulations with watchers, so we can conclude that our implementation is at least close enough to the state-of-the-art.